### PR TITLE
Add customizable wordSeparator and support alternatives with multiple words

### PR DESCRIPTION
**Pull request by tdjac0bs**
This PR allow you to customize the word separator.

- This allows encoding the space placement in sentences, like "Test ?" or "Test?" 
- This also allows to encode alternatives with multiple words like "A-B|A - B"

Details covered in issue #67

### CODE REVIEW

File: src/scripts/h5p-dictation-sentence.js [hunk 1]
```diff
@@ -406,16 +412,42 @@ class Sentence {
...
+      input = input.replaceAll(" ", this.params.wordSeparator);
+      input = input.replaceAll(Sentence.SPACE_ESCAPE, ' ');
+    }
+
+    // Add spaces to solution and break in parts
+    let wordsInput = input.trim() === '' ? [] : Sentence.addSpaces(input, this.params.autosplit, this.params.wordSeparator).split(this.params.wordSeparator).filter(word => word.length>0);
+    
+    // In case our wordSeparator is not space, we add spaces between tokens
+    if(' ' !== this.params.wordSeparator) {
+      wordsInput = wordsInput.map( (word,i) => i==wordsInput.length-1 ? word : word + ' ');
```
Comment (otacke) [inline src/scripts/h5p-dictation-sentence.js:449]: The H5P core team favors template-strings nowadays as Babel takes care of those for IE11 even without a polyfill.

File: semantics.json [hunk 2]
```diff
@@ -182,6 +182,14 @@
     "importance": "low",
     "description": "These options will let you control how the task behaves.",
     "fields": [
+      {
```
Comment (otacke) [inline semantics.json:185]: Could you please put this field after the `ignorePunctuation`? I try to keep these options in order of potential use, and using a different separator than a space is probably not the most likely one :-)

**Review by otacke — CHANGES_REQUESTED** [review]
Thanks a lot!

I have not yet run the code but given it a first look. I spotted some things that I could probably fix myself quicker than explaining them here, but of course I hope that you will contribute more things to H5P in the future :-)

Whenever you change `semantics.json`, you will also have to update the translation files that H5P bases these on. They're found in https://github.com/otacke/h5p-dictation/tree/master/language If you don't do that, the H5P editor will get messed up for other languages than English. You do not have to translate these, of course, but you will have to put in the things that can actually be translated. Here that'd be the label and the description of your new field.

You can do that either manually - duh - or use the [H5P CLI tool](https://h5p.org/h5p-cli-guide). For this case, you would add `"state": "new"` to the field in semantics.json and then run `h5p update-translations h5p-dictation` from within the directory that contains the source folder. Other values for the state property would be "updated" which would overwrite all corresponding strings in the translation files and "removed" which would delete the strings for a field.

Could you please also take a look at the [H5P core team's coding guidelines](https://h5p.org/documentation/for-developers/coding-guidelines) and stick to these? The project contains a [configuration file for ESLint](https://github.com/otacke/h5p-dictation/blob/master/.eslintrc.json). Not sure if you can include ESLint in your editor (using it in atom) - but it really helps if the editor gives you a hint that something is odd. You can run it manually as well, of course, but I have never tried if that's possible with plugins like "jest" that I use here - but why wouldn't that work ...

I have also left a couple of comments for you. Would be great if you took care of those.

**Comment by tdjac0bs** [comment]
Nice working on these changes.

**Comment by otacke** [comment]
@tdjac0bs Sorry for not answering earlier, but since I am not a native English speaker, I may have misinterpreted your answer. I was under the impression that your answer meant something like "I'll gladly work on these changes", but re-reading it, it might also mean "Have fun with doing that yourself". Could you please let me know what interpretation is closer to what you meant?

**Comment by tdjac0bs** [comment]
hello @otacke - you understood it correctly: "I'll gladly work on these changes"
working on this right now

File: src/scripts/h5p-dictation-sentence.js [hunk 3]
```diff
@@ -335,7 +337,11 @@ class Sentence {
    * @param {boolean} autosplit If true, automatically split respective symbols.
    * @return {string} Text with spaces and symbols.
    */
-  static addSpaces(text, options = {}) {
+  static addSpaces(text, options = {}, wordSeparator=' ') {
```
[the remark this answers is not in the document]
Comment (tdjac0bs) [reply src/scripts/h5p-dictation-sentence.js:365]: I think I missed something here, in the function definition the second argument is an object, and it's used like an object: https://prnt.sc/15yoe3t

However, at function call it's sending the `autoSplit` variable directly: https://prnt.sc/15yoimk

Can you clarify it?

[the remark this answers is not in the document]
Comment (otacke) [reply src/scripts/h5p-dictation-sentence.js:365]: Hmm, that feels like a bug in my end then. Please just keep the `wordSeparator` argument separate then. I'll deal with it when I check the options object. Thanks for pointing that out!

[the remark this answers is not in the document]
Comment (tdjac0bs) [reply src/scripts/h5p-dictation-sentence.js:365]: cool, thank you!

**Comment by tdjac0bs** [comment]
Hello, I've applied all suggested changes, but one error is happening.

Anything that I type in this new field is used as title for the accordion:

Text in new field:
image

Accordion title:
image

**Review by otacke — APPROVED** [review]
Something seems to have gone wrong with the translation files, but I'll fix that in a second myself.

The group label showing the symbols in the editor is not a bug, but a quirk of the H5P editor. It looks for the first (text) input field and uses its value as the group label. Don't know why, but that seems to be expected behavior.

**Pull request merged**